slices/reduce.go

```diff
@@ -0,0 +1,25 @@
+package slices
+
+import "golang.org/x/exp/constraints"
+
+func ReduceWith[S ~[]E, E any, A any](s S, acc A, fn func(A, E) A) A {
```

Comment:
`ReduceWithAggregator`?

Also is this not possible?
```suggestion
func ReduceWith[E any, A any](s []E, acc A, fn func(A, E) A) A {
```
Not sure if the compiler'll like it

Reply:
Also naming the vars something a bit more explicit can be useful
- s => elements
- acc => accumulator
- fn => transformationFunc
Or something? Otherwise we're really sending a telegram like we do in ruby

Reply:
what @Sypheos wrote is pretty standards, it allows typedef of slices to be accepted as paramenter,

for some reason you'd like:

```
type Strings []string
```

you could pass directly a value of type Strings as first param

Reply:
> * s => elements
> * acc => accumulator
> * fn => transformationFunc

I would go with something like:
- `s` => `slice`
- `acc` could stay the same, it is a pretty standard name for accumulators
- `fn` => `reduceFunc`